### PR TITLE
[BEAM-11770] bump google-cloud-pubsub to 1.110.3(google_cloud_platform_libraries_bom )

### PR DESCRIPTION
**Pull request by masahitojp**
bump google-cloud-pubsub to 1.110.3, because we follwed dependencies in google_cloud_platform_libraries_bom sets version.

The current master branch depends on `16.3.0` (#13990)

https://storage.googleapis.com/cloud-opensource-java-dashboard/com.google.cloud/libraries-bom/16.3.0/artifact_details.html

^ It depends on `com.google.cloud:google-cloud-pubsub:1.110.3`
Since this is newer than the current master(`1.108.6` ) , I think it's better to update

### results of ` sh sdks/java/build-tools/beam-linkage-check.sh origin/master beam-11770`

*Success*

https://gist.github.com/masahitojp/6a292e3d8ff5506383fca3a44805bafb

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by masahitojp** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by masahitojp** [comment]
Run Website PreCommit

**Comment by masahitojp** [comment]
R: @iemejia 
Could you review this PR?

**Comment by iemejia** [comment]
I am not familiar with the consequences of dependency upgrades for Google related dependencies, so in the future for google related dependencies please choose as a reviewer like @kennknowles @chamikaramj or @suztomo 

Can someone of you please help review this one or forward this to someone who can. Thanks!
Oh and thanks for your this and all the other recent contributions @masahitojp !

**Comment by suztomo** [comment]
Run Java Postcommit

**Comment by suztomo** [comment]
@masahitojp Would you try running the command below?

```
sh sdks/java/build-tools/beam-linkage-check.sh origin/master beam-11770
```

https://cwiki.apache.org/confluence/display/BEAM/Dependency+Upgrades#app-switcher (this has outdated command line parameter. It now takes 2 git branches.)

If there's no new conflict, it looks good to me (I'm not a committer though).

Note that there's back-and-forth (#13765 and #13972) of GCP dependencies in the last couple of weeks. They might interfere this PR; thank you for your patience.

**Comment by masahitojp** [comment]
Hi @suztomo 

I tried the command.
> sh sdks/java/build-tools/beam-linkage-check.sh origin/master beam-11770

but I got a below errors :cry: 
```
Failed to generate the baseline file. Check the build error above.
Checking linkage for origin/beam-11770
Previous HEAD position was 3930d12d47 Merge pull request #13981: [BEAM-11124] bump joda-time to 2.10.10
HEAD is now at 45ba4c5888 [BEAM-11770] bump google-cloud-pubsub to 1.111.0
Running linkage check (validate) for beam-sdks-java-core
using baseline build/linkagecheck/baseline-beam-sdks-java-core.xml
Error: Baseline file not found
Running linkage check (validate) for beam-sdks-java-io-google-cloud-platform
using baseline build/linkagecheck/baseline-beam-sdks-java-io-google-cloud-platform.xml
Error: Baseline file not found
Running linkage check (validate) for beam-runners-google-cloud-dataflow-java
 using baseline build/linkagecheck/baseline-beam-runners-google-cloud-dataflow-java.xml
Error: Baseline file not found
Running linkage check (validate) for beam-sdks-java-io-hadoop-format
 using baseline build/linkagecheck/baseline-beam-sdks-java-io-hadoop-format.xml
Error: Baseline file not found
```
Do you know what is causing it? (openjdk version "1.8.0_282")

**Comment by suztomo** [comment]
> Check the build error above.

Do you see any error above the line?

**Comment by suztomo** [comment]
I tried and got the same error. Digging further. https://gist.github.com/suztomo/b5a33361ad83c5e5c9054b393ac496c1

**Comment by suztomo** [comment]
@masahitojp It seems that beam-linkage-check.sh has not been migrated properly when this repository moved build script from Gradle to Kotlin. (Your observation helped me to identify the problem. Thank you!)

I raised a pull request to fix the problem https://github.com/apache/beam/pull/14006. Give me and Beam committers some moment.

**Comment by suztomo** [comment]
@masahitojp I rebased your branch and ran the tool

https://gist.github.com/suztomo/0ba02d93aa0b2f5d9701c50f7d7fd81f It seems that proto-google-cloud-pubsub-v1 needs to have a newer version. Would you check it?

**Comment by masahitojp** [comment]
@suztomo Thank you for your great contribution!!

> It seems that proto-google-cloud-pubsub-v1 needs to have a newer version. Would you check it?
I checked it. I can't say anything, but I think it's preferable to update it 

https://github.com/apache/beam/blob/b8dd9cb4dfbcacda3b50baa440a0f3642b230990/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L607

maybe　, In other words, I think it will affect this trend. 
> Note that there's back-and-forth (#13765 and #13972) of GCP dependencies in the last couple of weeks.

I think it would be good to put it on hold until this work is finished.  😢

**Comment by masahitojp** [comment]
Run Java PreCommit

**Comment by masahitojp** [comment]
The current master branch depends on 16.3.0 (#13990)
In other words, there was no longer any affect on us.

> Note that there's back-and-forth (#13765 and #13972) of GCP dependencies in the last couple of weeks.

#13990 is merged, and this PR is restarted, especially since it does not seem to be causing any problems in testing.

----
Now I've changed the purpose of this PR.

## what

> bump google-cloud-pubsub to 1.110.3, because we follwed dependencies in google_cloud_platform_libraries_bom sets version.


The current master branch depends on 16.3.0 (#13990)

https://storage.googleapis.com/cloud-opensource-java-dashboard/com.google.cloud/libraries-bom/16.3.0/artifact_details.html

^ It depends on com.google.cloud:google-cloud-pubsub:1.110.3
Since this is newer than the current master(1.108.6 ) , I think it's better to update

## why(background)

The apache beam specifies the versions of the following packages

* com.google.cloud:google-cloud-pubsub

But this package has dependencies with the following packages but this one is specified by `google_cloud_platform_libraries_bom `

* com.google.api.grpc:proto-google-cloud-pubsub-v1
* com.google.api.grpc:grpc-google-cloud-pubsub-v1

The dependencies of this package are difficult to understand for people who don't know about it, so there is a possibility that the dependencies will be broken unintentionally in the future. For this reason, I thought it would be better for me to unify them into `google_cloud_platform_libraries_bom `

^ It depends on com.google.cloud:google-cloud-pubsub:1.110.3
Since this is newer than the current master(1.108.6 ) , I think it's better to update

## results of sh sdks/java/build-tools/beam-linkage-check.sh origin/master beam-11770

Success

https://gist.github.com/masahitojp/6a292e3d8ff5506383fca3a44805bafb
(Thanks to your support @suztomo )


----

Could you re-review this PR?
R: @TheNeuralBit @iemejia

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:529]: This is a nice enhancement. No need to maintain google-cloud-pubsub separately. Thank you.

**Review by suztomo — APPROVED** [review]

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy
Comment (TheNeuralBit) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:529]: +1, thank you

**Review by TheNeuralBit — APPROVED** [review]

**Pull request merged**
